### ISSUE TITLE
pylance does not work depending on the location when receiving multiple inheritance of class

### ISSUE THREAD

**Issue by big100** [microsoft/pylance-release#3201]
## Environment data

-   Language Server version:  v2022.8.20
-   OS and version: 버전: 1.70.0 (user setup)
커밋: da76f93349a72022ca4670c1b84860304616aaa2
날짜: 2022-08-04T04:38:16.462Z
Electron: 18.3.5
Chromium: 100.0.4896.160
Node.js: 16.13.2
V8: 10.0.139.17-electron.0
OS: Windows_NT x64 10.0.19043

-   Python version (& distribution if applicable, e.g. Anaconda): 3.9.7

## Code Snippet

pylance does not work depending on the location when receiving multiple inheritance of class

![vscode_pylance버그](https://user-images.githubusercontent.com/37491415/184949603-b12c73a1-2c52-4951-9d43-4855f8c44cb0.gif)

others example : https://stackoverflow.com/questions/67242228/vscode-pylance-class-attributes-recognized-but-no-autocomplete-available

**Comment by judej** [comment]
Thanks for the report, would you please send a more complete code sample for us to debug?

thanks

**Comment by zocker-160** [comment]
@judej I have exactly the same issue, here some very basic example code:
[Pylance bugreport.zip](https://github.com/microsoft/pylance-release/files/9455039/Pylance.bugreport.zip)

`Ui_ui.py` is autogenerated code using `pyuic5`

**Comment by judej** [comment]
Thanks @zocker-160, will investigate the issue.

**Comment by bschnurr** [comment]
created a venv
pip install pyqt5

seem to get completions in your example
![image](https://user-images.githubusercontent.com/1946977/187554378-f66ea65c-d1e9-4923-88db-4b1dcfa2aa49.png)

**Comment by bschnurr** [comment]
can you provide logs? see https://github.com/microsoft/pylance-release/blob/main/TROUBLESHOOTING.md#filing-an-issue

**Comment by zocker-160** [comment]
@bschnurr thank you for looking into this, I think I found something that might be useful for you.

When I create a venv as you did in the screenshot and install PyQT5 inside, it works, so I can reproduce your situation.

However, I don't have PyQt5 installed in the venv but rather system wide and I have set `include-system-site-packages = true` inside `pyvenv.cfg`.

When I delete PyQt5 from the venv and enable system site packages, the suggestions disappear.
![Screenshot_20220831_025529](https://user-images.githubusercontent.com/36563538/187569513-bce09eaa-7139-46c0-9409-c77a2e44013b.png)

**Comment by bschnurr** [comment]
hmm. I guess we dont support `.cfg` environment configs files.

if the files dont exist in the environment folder, then we dont know about them.

I guess this is an enhancement request then.

**Comment by erictraut** [comment]
Pylance determines which import resolution paths it should use by running `python3` and executing a small script that dumps the value of `sys.path`.

@zocker-160, I would expect in your case that `sys.path` would include the site-packages directory for the system-wide Python environment. Please confirm by activating the venv, then running `python3` and typing `import sys` followed by `print(sys.path)`.

**Comment by zocker-160** [comment]
sure @erictraut here is my output inside the venv as seen in the screenshot above:
```
Pylance.Bugreport> source venv/bin/activate
(venv) Pylance.Bugreport> python3 --version
Python 3.10.4
(venv) Pylance.Bugreport> python3
Python 3.10.4 (main, Jun 29 2022, 12:14:53) [GCC 11.2.0] on linux
Type "help", "copyright", "credits" or "license" for more information.
>>> import sys
>>> print(sys.path)
['', '/usr/lib/python310.zip', '/usr/lib/python3.10', '/usr/lib/python3.10/lib-dynload', '/home/trashbox/Programmierkram/Ablage/python/Pylance.bugreport/venv/lib/python3.10/site-packages', '/usr/local/lib/python3.10/dist-packages', '/usr/lib/python3/dist-packages']
>>> 
```

**Comment by zocker-160** [comment]
I want to add, that when I set `include-system-site-packages = false` it does actually start complaining, that PyQt5 could not be resolved, so to me it looks like that the changes in the `.cfg` do have an effect:
![Screenshot_20220831_134231](https://user-images.githubusercontent.com/36563538/187670868-31f6d532-7ed4-4a87-a14d-3841a8da82d4.png)

**Comment by erictraut** [comment]
I don't see the `site-packages` directory of the system Python install in the `sys.path` that you provided above. Perhaps the library is installed in `dist-packages` instead? That directory is present in `sys.path`.

Let's see if we can confirm that theory. If you set `include-system-site-packages = true`, you indicated that the import was resolved (i.e. the yellow squiggles under `PyQt5.QtWidgets` no longer appears). If you right-click on `QMainWindow` in the import statement and select "Go To Declaration", where does it take you? This will tell us which directory PyQt5 is contained within. If my theory is correct, it will be in the `dist-packages` directory.

**Comment by zocker-160** [comment]
@erictraut  yes PyQt5 installed from the package manager is placed inside `dist-packages`

https://user-images.githubusercontent.com/36563538/187692902-7b829869-4a7c-48a1-9576-2e6326b0d67a.mp4

**Comment by rchiodo** [comment]
Do we perhaps cache the sys path somewhere (and it needs to be cleaned)? This is working fine for me with the same setup.


![qt5button](https://user-images.githubusercontent.com/19672699/195953898-08764f55-0b40-4419-8ed0-ebc39a84f285.gif)

**Comment by rchiodo** [comment]
@zocker-160 can you provide the logs as outlined here:
https://github.com/microsoft/pylance-release/blob/main/TROUBLESHOOTING.md#filing-an-issue

**Comment by zocker-160** [comment]
@rchiodo sure no problerm, here you go
[pylance-trace.log](https://github.com/microsoft/pylance-release/files/9809547/pylance-trace.log)

**Comment by rchiodo** [comment]
Looks like it's indexing the correct pyi files. Maybe it's just too slow.

If you add this:

![image](https://user-images.githubusercontent.com/19672699/196490837-ac1ea679-d615-4559-be0b-22da2c31f7e7.png)

And goto def on the `acceptDrops` does it find the definition in the dist-packages?

If not can you set this setting, restart VS code, and upload a log again:

```json
  "python.analysis.logLevel": "Trace",
```

**Comment by rchiodo** [comment]
Might also be a problem with the pyi file. Here's what my QTWidgets.pyi looks like:

[QtWidgets.pyi.txt](https://github.com/microsoft/pylance-release/files/9812805/QtWidgets.pyi.txt)

**Comment by zocker-160** [comment]
video and the resulting log

https://user-images.githubusercontent.com/36563538/196497498-847fff47-32eb-4314-bdd8-ac329a97aa97.mp4

[pylance-trance2.log](https://github.com/microsoft/pylance-release/files/9813517/pylance-trance2.log)

**Comment by rchiodo** [comment]
The log doesn't seem to be at log level "trace". It should show a lot more data. Please double check you have this in your user settings.json:

```json
  "python.analysis.logLevel": "Trace",
```

You can also set it here:

![image](https://user-images.githubusercontent.com/19672699/196499060-001c6161-02e2-4074-b798-f252923ddf08.png)


If you goto def on PyQt5.QtWidgets, where does that go? And if that file opens, does pylance show any errors in it?

**Comment by zocker-160** [comment]
alright so just to be sure, nothing on my system might cause this problem, I set up a completely fresh VM with VScode installed and the code example from above:

https://user-images.githubusercontent.com/36563538/196503902-52c90b89-91e1-4597-8c02-1e8108d03f01.mp4

[pylance-trace3.log](https://github.com/microsoft/pylance-release/files/9813744/pylance-trace3.log)

I did set the log level to trace, as you can see in the clip.

**Comment by zocker-160** [comment]
I don't think the `PyQt5.QtWidgets.pyi` is at fault here, because I can autocomplete the same button correctly inside `Ui_ui.py`

**Comment by rchiodo** [comment]
Thanks. I'm at a loss right now. I'll have to generate a build that has more logging and post it here.

**Comment by heejaechang** [comment]
@big100 can you provide us actual files to repro the issue? without that, it will be hard for us to figure out.

**Comment by rchiodo** [comment]
@zocker-160 if you can e-mail me (same name as my github id with an `at` microsoft.com), I can send you a pylance extension with extra logging.

**Comment by zocker-160** [comment]
> @big100 can you provide us actual files to repro the issue? without that, it will be hard for us to figure out.

I have uploaded my example project (that you can see above and that I use for demonstrating the issue) here: https://github.com/microsoft/pylance-release/issues/3201#issuecomment-1231940084

> if you can e-mail me (same name as my github id with an at microsoft.com), I can send you a pylance extension with extra logging.

I will, I highly appreciate your efforts to find the cause of this issue, thank you.

**Comment by zocker-160** [comment]
I loaded the same example code using the version provided by @rchiodo:

[pylance-trace_new.log](https://github.com/microsoft/pylance-release/files/9823338/pylance-trace_new.log)

I also want to point out, that as OP switching the order of the inheritance does suddenly bring the suggestions back:

https://user-images.githubusercontent.com/36563538/196777762-5b26653d-44ca-4099-b215-0370d10caed6.mp4

**Comment by rchiodo** [comment]
Thanks that helps. At least it's narrowed where the problem is. I'll likely need to send you another vsix with even more logging. 

We're unable to find the symbols for the `a_crazy_button` member. Which is probably a failure to find the symbols for the class `MainWindow`, but I'll have to add more logging to be sure.

**Comment by rchiodo** [comment]
Figured out the root cause. The QtGui.pyi in version PyQt5 5.15.6 is wrong. This line here:

```python
class QPaintDevice(sip.simplewrapper):
```

should say this:

```python
class QPaintDevice(PyQt5.sip.simplewrapper):
```

This was fixed in 5.15.7 (which is the latest).
